### PR TITLE
CLOUDP-76088: Disable Realm and Charts icons in MongoNav

### PR DESCRIPTION
**Pull request by JamesHeppenstall**
## ✍️ Proposed changes

As part of [this epic](https://jira.mongodb.org/browse/CLOUDP-55272), we want to disable the Realm and Charts icons in `MongoNav` when an Atlas project is designated as an China regions-only project. This WIP PR adds the necessary logic to achieve this. See the associated PR for [CLOUDP-76088](https://github.com/10gen/mms/pull/35481) for the changes to the endpoint `/user/shared` from which `MongoNav` gets information about the current project.

🎟 _Jira ticket:_ [CLOUDP-76088](https://jira.mongodb.org/browse/CLOUDP-76088)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

## 💬 Further comments

Here's what the icons and tooltip look like when the current project is an China regions-only project:
Screen Shot 2020-12-01 at 1 09 46 PM

### CODE REVIEW

**Comment by lazytype** [comment]
From testing this locally I noticed a weird behavior. Clicking one of the tabs briefly displays the cursor as `cursor: pointer` and causes the tooltip disappear and reappear

**Comment by lazytype** [comment]
Pressing the Enter key when one of the tabs is focused still navigates to a new page.

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx [hunk 1]
```diff
@@ -296,6 +322,11 @@ export default function ProjectNav({
   const isCloudManager = current?.planType === PlanType.Cloud;
   const isLoading = !!current;
   const currentProjectId = current?.projectId;
+  const useCNRegionsOnly =
```
Comment (lazytype) [inline packages/mongo-nav/src/project-nav/ProjectNav.tsx:325]: could we do something like the following to avoid the duplicated product logic below
```ts
const useCNRegionsOnly = current?.useCNRegionsOnly ?? false;
const planType = current?.planType;

const isCNRegionsOnly = useCallback(
  (product: Product) => {
    const disabledProducts: Array<Product> = [Product.Realm, Product.Charts];

    return (
      useCNRegionsOnly &&
      planType === PlanType.Atlas &&
      disabledProducts.includes(product)
    );
  },
  [useCNRegionsOnly, planType],
);
```

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx [hunk 2]
```diff
@@ -442,43 +477,69 @@ export default function ProjectNav({
...
 
             <li role="none" className={productTabStyle}>
-              <a
-                data-testid="project-nav-charts"
-                href={projectNav.charts}
-                className={getProductClassName('charts')}
-                aria-disabled={!current}
-                tabIndex={current ? 0 : -1}
-                onClick={onElementClick(ProjectNavCharts)}
+              <Tooltip
```
Comment (lazytype) [inline packages/mongo-nav/src/project-nav/ProjectNav.tsx:513]: Can we pull out the duplicated tooltip logic into a component? e.g.
```tsx
  const WithCNRegionsOnlyTooltip = useCallback(
    ({ children }: { children: React.ReactElement }) => {
      return (
        <Tooltip
          {...sharedTooltipProps}
          tabIndex={-1}
          align="bottom"
          justify="middle"
          usePortal={false}
          darkMode={true}
          enabled={useCNRegionsOnly}
          trigger={children}
        >
          {useCNRegionsOnlyMessage}
        </Tooltip>
      );
    },
    [sharedTooltipProps, useCNRegionsOnly],
  );

  ...

  <WithCNRegionsOnlyTooltip>
    <a
      data-testid="project-nav-charts"
      href={useCNRegionsOnly ? hosts.cloud : projectNav.charts}
      className={getProductClassName('charts')}
      aria-disabled={!current}
      tabIndex={current ? 0 : -1}
      onClick={onElementClick(ProjectNavCharts)}
    >
      {!isMobile && (
        <ChartsIcon
          {...productIconProp.prop}
          className={getIconStyle(Product.Charts)}
          active={activeProduct === Product.Charts && isLoading}
        />
      )}
      Charts
    </a>
  </WithCNRegionsOnlyTooltip>
```

File: packages/mongo-nav/src/types.ts [hunk 3]
```diff
@@ -203,6 +203,7 @@ export interface CurrentProjectInterface extends ProjectInterface {
   alertsOpen: number;
   chartsActivated: boolean;
   status?: ProjectStatus;
+  useCNRegionsOnly: boolean;
```
Comment (lazytype) [inline packages/mongo-nav/src/types.ts:206]: would `isCNRegionsOnly` make more sense?

**Review by lazytype — CHANGES_REQUESTED** [review]

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 1 above)
Comment (JamesHeppenstall) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:325]: 👍

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 2 above)
Comment (JamesHeppenstall) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:513]: 👍

File: packages/mongo-nav/src/types.ts (hunk 3 above)
Comment (JamesHeppenstall) [reply packages/mongo-nav/src/types.ts:206]: I agree that `isCNRegionsOnly` might make more sense but the field name on the project in the Atlas backend is called `useCNRegionsOnly` so I wanted to keep things consistent.

**Comment by JamesHeppenstall** [comment]
@lazytype thanks for the initial review. I addressed all your comments and also decided to only include the anchor tag if `useCNRegionsOnly` is false, which prevents some of the weird behavior you saw. Let me know what you think!

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx [hunk 4]
```diff
@@ -442,43 +524,51 @@ export default function ProjectNav({
...
-              </a>
+              {useCNRegionsOnly && (
+                <WithCNRegionsOnlyTooltip>
+                  <div className={getProductClassName(Product.Realm)}>
+                    {getRealmIcon()}
+                    Realm
+                  </div>
+                </WithCNRegionsOnlyTooltip>
+              )}
+              {!useCNRegionsOnly && (
```
Comment (lazytype) [inline packages/mongo-nav/src/project-nav/ProjectNav.tsx:535]: hm, is there a way we can get both branches to be defined in `WithCNRegionsOnlyTooltip` ?

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx [hunk 5]
```diff
@@ -247,6 +268,13 @@ type ProjectNavProps = Pick<
   | 'environment'
   | 'alertPollingInterval'
 > & {
+  admin?: boolean;
+  activeProduct?: Product;
+  activeNav?: ActiveNavElement;
+  onProjectChange?: (onChangeInterface: OnChangeInterface) => void;
+  mode?: Mode;
+  alertPollingInterval?: number;
+  environment?: Environment;
```
Comment (lazytype) [inline packages/mongo-nav/src/project-nav/ProjectNav.tsx:277]: curious about why this change is needed. does the `Pick` above not already include these for us?

**Review by lazytype — COMMENTED** [review]
could we also add a couple of test cases to cover the new behaviors:
- clicking on the tabs when `useCNRegionsOnly` is `true` doesn't have any effect, e.g. `expect(getByText('Realm').closest('a')).toBeNull()`
- hovering over the tabs makes the tooltip visible

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 4 above)
Comment (JamesHeppenstall) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:535]: It doesn't seem to render properly when I put both branches in `WithCNRegionsOnlyTooltip`. Also, just wondering why we want to render the tooltip if `useCNRegionsOnly == false`?

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 5 above)
Comment (JamesHeppenstall) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:277]: I thought the same thing but `yarn fix` started throwing errors about those props not being defined.

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 4 above)
Comment (lazytype) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:535]: I was thinking the semantics of it would be more like `WithTooltipIfCNRegionsOnly` and handle the branching so we don't need to repeat it for Realm and Charts

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 5 above)
Comment (lazytype) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:277]: ah, seems like a linter false-positive. you can wrap those lines in
```
/* eslint-disable react/prop-types */
...
/* eslint-enable react/prop-types */
```

File: packages/mongo-nav/src/project-nav/ProjectNav.tsx (hunk 4 above)
Comment (JamesHeppenstall) [reply packages/mongo-nav/src/project-nav/ProjectNav.tsx:535]: Ah got it 😄 I'll update the PR to handle this

**Comment by JamesHeppenstall** [comment]
@lazytype this is RFAL 😄

**Review by lazytype — APPROVED** [review]
looks great!
